src/operations.cc

```diff
@@ -351,6 +351,31 @@ namespace sharp {
     }
   }
 
+  /*
+   * Unflatten
+   */
+  VImage Unflatten(VImage image, std::vector<double> const thresholds) {
+    if (HasAlpha(image)) {
+      throw VError("Cannot unflatten an image that already has an alpha channel");
+    }
+    size_t const bands = static_cast<size_t>(image.bands());
+    if (thresholds.size() != 1 && thresholds.size() != bands) {
+      throw VError("band / threshold count mismatch");
```

Comment:
Perhaps add something to this message about the error originating from `unflatten`. Is it worth testing this path also?